cpu/cc26x0/periph/rfc.c

```diff
@@ -72,13 +58,54 @@ void rfc_irq_disable(void)
     NVIC_EnableIRQ(RF_HW_IRQN);
 }
 
+uint32_t rfc_send_cmd(uint32_t ropAddr)
+{
+    RFC_DBELL->CMDR = ropAddr;
+  /* wait for cmd ack (rop cmd was submitted successfully) */
+  while(RFC_DBELL->RFACKIFG<<31);
+  while(!RFC_DBELL->CMDSTA);
+  if (RFC_DBELL->CMDSTA == 1)
+  {
+    printf("%s: radio operation command acknowledged. \n",__FUNCTION__);
+  }
+  else
+  {
+    printf("%s: acknowledgement of radio operation command failed (0x%" PRIu32 ") \n",__FUNCTION__,RFC_DBELL->CMDSTA);
+  }
+
+  return RFC_DBELL->CMDSTA;
+}
+
+uint16_t rfc_wait_cmd_done(void* ropCmd)
```

Comment:
I agree that the current implementation is really ugly and that we should hold on before making anything "official". The thing is, I faced many strange timing issue regarding the different registers: for example the ack interrupt would trigger, but the CMDSTA would not be available yet. So the two while are somewhat redundant yet it was the only way to make things work.

I tried to use the interrupt service routine, but I couldn't find a way to pause the `rfc_send_cmd()` until `isr_rfc_cmd_ack` has been executed... I'll think about this and run some experiment tomorrow. I agree that we should find a more elegant way to check for the different states of execution of a command.